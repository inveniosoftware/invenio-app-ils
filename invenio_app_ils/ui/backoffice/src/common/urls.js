export const URLS = {
  ROOT: '/',
  BASENAME: '/backoffice',
  ITEM_LIST: '/items',
  ITEM_DETAILS: id => `/items/${id}`,
  LOAN_LIST: '/loans',
  LOAN_DETAILS: id => `/loans/${id}`,
};
