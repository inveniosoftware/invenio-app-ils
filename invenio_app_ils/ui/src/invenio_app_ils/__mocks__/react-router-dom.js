export default {
  Link: jest.fn(),
};
