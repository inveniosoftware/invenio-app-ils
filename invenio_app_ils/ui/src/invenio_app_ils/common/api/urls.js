export const ApiURLS = {
  loans: {
    loan: '/circulation/loans/:loanPid',
    list: '/circulation/loans/',
    checkout: '/circulation/loans/checkout',
    request: '/circulation/loans/request',
  },
  items: {},
  documents: {},
  locations: {},
  users: {},
};
