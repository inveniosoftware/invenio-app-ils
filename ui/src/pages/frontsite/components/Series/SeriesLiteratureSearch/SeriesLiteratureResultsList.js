import React, { Component } from 'react';
import { Item } from 'semantic-ui-react';
import { recordToPidType } from '@api/utils';
import { DocumentListEntry } from '@pages/frontsite/Documents/DocumentsSearch/DocumentListEntry';
import { SeriesListEntry } from '@pages/frontsite/Documents/DocumentsSearch/SeriesListEntry';
import { findVolume } from '..';
import { ResultsList } from 'react-searchkit';

export default class DocumentSearchResultsList extends Component {
  renderResultsList = results => {
    return results.length ? (
      <Item.Group>
        {results.map(result => {
          const volume = findVolume(
            result,
            result.metadata ? result.metadata.pid : null
          );
          return recordToPidType(result) === 'docid' ? (
            <DocumentListEntry
              key={result.metadata.pid}
              data-test={result.metadata.pid}
              metadata={result.metadata}
              volume={volume}
            />
          ) : (
            <SeriesListEntry
              key={result.metadata.pid}
              data-test={result.metadata.pid}
              metadata={result.metadata}
              volume={volume}
            />
          );
        })}
      </Item.Group>
    ) : null;
  };

  render() {
    return <ResultsList renderElement={this.renderResultsList} />;
  }
}
