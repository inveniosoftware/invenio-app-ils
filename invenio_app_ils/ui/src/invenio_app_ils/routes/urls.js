import { generatePath } from 'react-router-dom';
import { invenioConfig } from '../common/config';

const FrontSiteBase = '/';

const FrontSiteRoutesList = {
  home: FrontSiteBase,
  patronProfile: `${FrontSiteBase}profile`,
  documentsList: `${FrontSiteBase}search`,
  documentDetails: `${FrontSiteBase}records/:documentPid`,
};

const FrontSiteRoutesGenerators = {
  documentsListWithQuery: qs => `${FrontSiteRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(FrontSiteRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
};

export const FrontSiteRoutes = {
  ...FrontSiteRoutesGenerators,
  ...FrontSiteRoutesList,
};

const BackOfficeBase = '/backoffice';

const BackOfficeRoutesList = {
  home: BackOfficeBase,
  documentsList: `${BackOfficeBase}/documents`,
  documentsCreate: `${BackOfficeBase}/documents/create`,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  eitemsList: `${BackOfficeBase}/eitems`,
  eitemDetails: `${BackOfficeBase}/eitems/:eitemPid`,
  itemsList: `${BackOfficeBase}/items`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  loansList: `${BackOfficeBase}/loans`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  patronsList: `${BackOfficeBase}/patrons`,
  patronDetails: `${BackOfficeBase}/patrons/:patronPid`,
  locationsList: `${BackOfficeBase}/locations`,
  seriesList: `${BackOfficeBase}/series`,
  seriesDetails: `${BackOfficeBase}/series/:seriesPid`,
};

const BackOfficeRouteGenerators = {
  documentsListWithQuery: qs => `${BackOfficeRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(BackOfficeRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
  eitemDetailsFor: eitemPid =>
    generatePath(BackOfficeRoutesList.eitemDetails, { eitemPid: eitemPid }),
  itemsListWithQuery: qs => `${BackOfficeRoutesList.itemsList}?q=${qs}`,
  itemDetailsFor: itemPid =>
    generatePath(BackOfficeRoutesList.itemDetails, { itemPid: itemPid }),
  loansListWithQuery: qs => `${BackOfficeRoutesList.loansList}?q=${qs}`,
  loanDetailsFor: loanPid =>
    generatePath(BackOfficeRoutesList.loanDetails, { loanPid: loanPid }),
  patronDetailsFor: patronPid =>
    generatePath(BackOfficeRoutesList.patronDetails, { patronPid: patronPid }),
  seriesListWithQuery: qs => `${BackOfficeRoutesList.seriesList}?q=${qs}`,
  seriesDetailsFor: seriesPid =>
    generatePath(BackOfficeRoutesList.seriesDetails, { seriesPid: seriesPid }),
};

export const BackOfficeRoutes = {
  ...BackOfficeRoutesList,
  ...BackOfficeRouteGenerators,
};

export const openRecordEditor = (path, recid = '') => {
  window.open(`${invenioConfig.editor.url}${path}${recid}`, '_blank');
};
