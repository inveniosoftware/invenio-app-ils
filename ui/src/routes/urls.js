import { generatePath } from 'react-router-dom';

const AuthenticationBase = '/';

const AuthenticationRoutesList = {
  login: `${AuthenticationBase}login`,
  redirectUrlAfterLogin: `${AuthenticationBase}login?next=:nextUrl`,
  confirmEmail: `${AuthenticationBase}confirm-email`,
};

const AuthenticationRoutesGenerators = {
  redirectAfterLogin: nextUrl =>
    generatePath(AuthenticationRoutesList.redirectUrlAfterLogin, {
      nextUrl: nextUrl,
    }),
};

export const AuthenticationRoutes = {
  ...AuthenticationRoutesList,
  ...AuthenticationRoutesGenerators,
};

const FrontSiteBase = '/';

const FrontSiteRoutesList = {
  home: FrontSiteBase,
  patronProfile: `${FrontSiteBase}profile`,
  documentsList: `${FrontSiteBase}search`,
  documentDetails: `${FrontSiteBase}literature/:documentPid`,
  documentRequestForm: `${FrontSiteBase}request`,
  seriesDetails: `${FrontSiteBase}series/:seriesPid`,
};

const FrontSiteRoutesGenerators = {
  documentsListWithQuery: qs => `${FrontSiteRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(FrontSiteRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
  seriesDetailsFor: seriesPid =>
    generatePath(FrontSiteRoutesList.seriesDetails, {
      seriesPid: seriesPid,
    }),
};

export const FrontSiteRoutes = {
  ...FrontSiteRoutesGenerators,
  ...FrontSiteRoutesList,
};

const BackOfficeBase = '/backoffice';

const BackOfficeRoutesList = {
  home: BackOfficeBase,
  documentCreate: `${BackOfficeBase}/documents/create`,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  documentEdit: `${BackOfficeBase}/documents/:documentPid/edit`,
  documentRequestDetails: `${BackOfficeBase}/document-requests/:documentRequestPid`,
  documentRequestsList: `${BackOfficeBase}/document-requests`,
  documentsList: `${BackOfficeBase}/documents`,
  eitemCreate: `${BackOfficeBase}/eitems/create`,
  eitemDetails: `${BackOfficeBase}/eitems/:eitemPid`,
  eitemEdit: `${BackOfficeBase}/eitems/:eitemPid/edit`,
  eitemsList: `${BackOfficeBase}/eitems`,
  ilocationsCreate: `${BackOfficeBase}/internal-locations/create`,
  ilocationsEdit: `${BackOfficeBase}/internal-locations/:ilocationPid/edit`,
  itemCreate: `${BackOfficeBase}/items/create`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  itemEdit: `${BackOfficeBase}/items/:itemPid/edit`,
  itemsList: `${BackOfficeBase}/items`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  loansList: `${BackOfficeBase}/loans`,
  locationsCreate: `${BackOfficeBase}/locations/create`,
  locationsEdit: `${BackOfficeBase}/locations/:locationPid/edit`,
  locationsList: `${BackOfficeBase}/locations`,
  patronDetails: `${BackOfficeBase}/patrons/:patronPid`,
  patronsList: `${BackOfficeBase}/patrons`,
  seriesCreate: `${BackOfficeBase}/series/create`,
  seriesDetails: `${BackOfficeBase}/series/:seriesPid`,
  seriesEdit: `${BackOfficeBase}/series/:seriesPid/edit`,
  seriesList: `${BackOfficeBase}/series`,
  stats: {
    home: `${BackOfficeBase}/stats`,
  },
};

const BackOfficeRouteGenerators = {
  documentEditFor: documentPid =>
    generatePath(BackOfficeRoutesList.documentEdit, {
      documentPid: documentPid,
    }),
  documentsListWithQuery: qs => `${BackOfficeRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(BackOfficeRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
  documentRequestsListWithQuery: qs =>
    `${BackOfficeRoutesList.documentRequestsList}?q=${qs}`,
  documentRequestDetailsFor: documentRequestPid =>
    generatePath(BackOfficeRoutesList.documentRequestDetails, {
      documentRequestPid: documentRequestPid,
    }),
  eitemDetailsFor: eitemPid =>
    generatePath(BackOfficeRoutesList.eitemDetails, { eitemPid: eitemPid }),
  eitemEditFor: eitemPid =>
    generatePath(BackOfficeRoutesList.eitemEdit, { eitemPid: eitemPid }),
  eItemsListWithQuery: qs => `${BackOfficeRoutesList.eitemsList}?q=${qs}`,
  itemsListWithQuery: qs => `${BackOfficeRoutesList.itemsList}?q=${qs}`,
  itemDetailsFor: itemPid =>
    generatePath(BackOfficeRoutesList.itemDetails, { itemPid: itemPid }),
  itemEditFor: itemPid =>
    generatePath(BackOfficeRoutesList.itemEdit, { itemPid: itemPid }),
  loansListWithQuery: qs => `${BackOfficeRoutesList.loansList}?q=${qs}`,
  loanDetailsFor: loanPid =>
    generatePath(BackOfficeRoutesList.loanDetails, { loanPid: loanPid }),
  ilocationsEditFor: ilocationPid =>
    generatePath(BackOfficeRoutesList.ilocationsEdit, {
      ilocationPid: ilocationPid,
    }),
  locationsEditFor: locationPid =>
    generatePath(BackOfficeRoutesList.locationsEdit, {
      locationPid: locationPid,
    }),
  patronDetailsFor: patronPid =>
    generatePath(BackOfficeRoutesList.patronDetails, { patronPid: patronPid }),
  seriesListWithQuery: qs => `${BackOfficeRoutesList.seriesList}?q=${qs}`,
  seriesDetailsFor: seriesPid =>
    generatePath(BackOfficeRoutesList.seriesDetails, { seriesPid: seriesPid }),
  seriesEditFor: seriesPid =>
    generatePath(BackOfficeRoutesList.seriesEdit, {
      seriesPid: seriesPid,
    }),
};

export const BackOfficeRoutes = {
  ...BackOfficeRoutesList,
  ...BackOfficeRouteGenerators,
};

const AcquisitionBase = `${BackOfficeBase}/acquisition`;

const AcquisitionRoutesList = {
  orderCreate: `${AcquisitionBase}/orders/create`,
  orderDetails: `${AcquisitionBase}/orders/:orderPid`,
  orderEdit: `${AcquisitionBase}/orders/:orderPid/edit`,
  ordersList: `${AcquisitionBase}/orders`,
  vendorCreate: `${AcquisitionBase}/vendors/create`,
  vendorDetails: `${AcquisitionBase}/vendors/:vendorPid`,
  vendorEdit: `${AcquisitionBase}/vendors/:vendorPid/edit`,
  vendorsList: `${AcquisitionBase}/vendors`,
};

const AcquisitionRouteGenerators = {
  orderDetailsFor: orderPid =>
    generatePath(AcquisitionRoutesList.orderDetails, { orderPid: orderPid }),
  orderEditFor: orderPid =>
    generatePath(AcquisitionRoutesList.orderEdit, { orderPid: orderPid }),
  ordersListWithQuery: qs => `${AcquisitionRoutesList.ordersList}?q=${qs}`,
  vendorDetailsFor: vendorPid =>
    generatePath(AcquisitionRoutesList.vendorDetails, { vendorPid: vendorPid }),
  vendorEditFor: vendorPid =>
    generatePath(AcquisitionRoutesList.vendorEdit, { vendorPid: vendorPid }),
  vendorsListWithQuery: qs => `${AcquisitionRoutesList.vendorsList}?q=${qs}`,
};

export const AcquisitionRoutes = {
  ...AcquisitionRoutesList,
  ...AcquisitionRouteGenerators,
};

const ILLBase = `${BackOfficeBase}/ill`;

const ILLRoutesList = {
  libraryList: `${ILLBase}/libraries`,
  libraryDetails: `${ILLBase}/libraries/:libraryPid`,
  libraryEdit: `${ILLBase}/libraries/:libraryPid/edit`,
  libraryCreate: `${ILLBase}/libraries/create`,
  borrowingRequestList: `${ILLBase}/borrowing-requests`,
  borrowingRequestDetails: `${ILLBase}/borrowing-requests/:borrowingRequestPid`,
  borrowingRequestEdit: `${ILLBase}/borrowing-requests/:borrowingRequestPid/edit`,
  borrowingRequestCreate: `${ILLBase}/borrowing-requests/create`,
};

const ILLRoutesGenerators = {
  libraryDetailsFor: libraryPid =>
    generatePath(ILLRoutesList.libraryDetails, {
      libraryPid: libraryPid,
    }),
  libraryEditFor: libraryPid =>
    generatePath(ILLRoutesList.libraryEdit, {
      libraryPid: libraryPid,
    }),
  documentRequestsListWithQuery: qs =>
    `${BackOfficeRoutesList.documentRequestsList}?q=${qs}`,
  borrowingRequestDetailsFor: borrowingRequestPid =>
    generatePath(ILLRoutesList.borrowingRequestDetails, {
      borrowingRequestPid: borrowingRequestPid,
    }),
  borrowingRequestEditFor: borrowingRequestPid =>
    generatePath(ILLRoutesList.borrowingRequestEdit, {
      borrowingRequestPid: borrowingRequestPid,
    }),
};

export const ILLRoutes = {
  ...ILLRoutesList,
  ...ILLRoutesGenerators,
};

export const DetailsRouteByPidTypeFor = pidType => {
  switch (pidType) {
    case 'pitmid':
      return BackOfficeRouteGenerators.itemDetailsFor;
    case 'illbid':
      return ILLRoutesGenerators.borrowingRequestDetailsFor;
    default:
      throw new Error(
        `Cannot generate url to the detail page for unknown pidType: ${pidType}`
      );
  }
};
