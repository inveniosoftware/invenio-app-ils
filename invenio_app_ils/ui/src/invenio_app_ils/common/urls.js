import { generatePath } from 'react-router-dom';
import { loan as loanApi } from './api/loan';
import { item as itemApi } from './api/item';

export const FrontSiteURLS = {};

const BackOfficeBase = '/backoffice';

export const BackOfficeURLS = {
  home: BackOfficeBase,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  itemsSearch: `${BackOfficeBase}/items`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  loansSearch: `${BackOfficeBase}/loans`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  patronDetails: `${BackOfficeBase}/users/:userPid`,
};

export const viewLoanDetailsUrl = loanPid => {
  return generatePath(BackOfficeURLS.loanDetails, {
    loanPid: loanPid,
  });
};

export const showAllLoansUrl = (
  documentPid,
  itemPid,
  state,
  patronPid,
  extraQuery
) => {
  const qs = loanApi.buildLoansQuery(
    documentPid,
    itemPid,
    state,
    patronPid,
    extraQuery
  );
  return `${BackOfficeURLS.loansSearch}?q=${qs}`;
};

export const viewItemDetailsUrl = itemPid => {
  return generatePath(BackOfficeURLS.itemDetails, {
    itemPid: itemPid,
  });
};

export const showAllItemsUrl = (documentPid, state, extraQuery) => {
  const qs = itemApi.buildItemsQuery(documentPid, state, extraQuery);
  return `${BackOfficeURLS.itemsSearch}?q=${qs}`;
};
