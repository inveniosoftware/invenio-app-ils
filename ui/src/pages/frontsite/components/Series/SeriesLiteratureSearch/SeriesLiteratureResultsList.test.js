import React from 'react';
import { mount, shallow } from 'enzyme';
import documentsTestData from '@testData/documents.json';
import seriesTestData from '@testData/series.json';
import { SeriesLiteratureResultsList } from './SeriesLiteratureResultsList';
import { BrowserRouter } from 'react-router-dom';
import { FrontSiteRoutes } from '@routes/urls';

const series = seriesTestData[0];

const results = {
  total: 4,
  hits: [
    { metadata: documentsTestData[0] },
    { metadata: documentsTestData[1] },
    { metadata: seriesTestData[1] },
    { metadata: seriesTestData[2] },
  ],
};

describe('SeriesLiteratureResultsList tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  it('should load the SeriesLiteratureResultsList component', () => {
    const component = shallow(
      <SeriesLiteratureResultsList metadata={series} results={results.hits} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should link to document and series details', () => {
    component = mount(
      <BrowserRouter>
        <SeriesLiteratureResultsList metadata={series} results={results.hits} />
      </BrowserRouter>
    );

    // Add serial relation to hits
    let volume = 1;
    for (const hit of results.hits) {
      hit.metadata.relations = {
        serial: {
          pid: series.pid,
          pid_type: 'serid',
          relation_type: 'serial',
          title: `Test title vol. ${volume}`,
          volume: volume + 's',
        },
      };
      volume++;
    }

    const documentEntries = component.find('DocumentListEntry');
    const seriesEntries = component.find('SeriesListEntry');
    expect(documentEntries).toHaveLength(2);
    expect(seriesEntries).toHaveLength(2);

    const firstDocId = results.hits[0].metadata.pid;
    const documentLink = documentEntries
      .filterWhere(element => element.prop('data-test') === firstDocId)
      .find('Link')
      .first();

    const firstSeriesId = results.hits[2].metadata.pid;
    const seriesLink = seriesEntries
      .filterWhere(element => element.prop('data-test') === firstSeriesId)
      .find('Link')
      .first();

    expect(documentLink.props().to).toEqual(
      FrontSiteRoutes.documentDetailsFor(firstDocId)
    );
    expect(seriesLink.props().to).toEqual(
      FrontSiteRoutes.seriesDetailsFor(firstSeriesId)
    );
  });
});
