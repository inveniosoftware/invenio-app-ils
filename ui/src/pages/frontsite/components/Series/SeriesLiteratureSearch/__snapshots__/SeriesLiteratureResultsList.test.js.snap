// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesLiteratureResultsList tests should load the SeriesLiteratureResultsList component 1`] = `
<ItemGroup>
  <DocumentListEntry
    data-test="docid-1"
    key="docid-1"
    metadata={
      Object {
        "$schema": "https://127.0.0.1:5000/schemas/documents/document-v1.0.0.json",
        "abstract": "This is an abstract",
        "authors": Array [
          Object {
            "full_name": "Jack E. Davis",
          },
        ],
        "document_type": "BOOK",
        "edition": "1",
        "keywords": Object {
          "source": "X",
          "value": "Patata",
        },
        "languages": Array [
          "en",
        ],
        "pid": "docid-1",
        "publication_year": "1950",
        "title": "The Gulf: The Making of An American Sea",
      }
    }
    volume={null}
  />
  <DocumentListEntry
    data-test="docid-2"
    key="docid-2"
    metadata={
      Object {
        "$schema": "https://127.0.0.1:5000/schemas/documents/document-v1.0.0.json",
        "abstract": "This is an abstract",
        "authors": Array [
          Object {
            "full_name": "Caroline Fraser",
          },
        ],
        "document_type": "BOOK",
        "keywords": Object {
          "source": "X",
          "value": "Patata",
        },
        "languages": Array [
          "it",
        ],
        "pid": "docid-2",
        "publication_year": "1950",
        "title": "Prairie Fires: The American Dreams of Laura Ingalls Wilder",
      }
    }
    volume={null}
  />
  <SeriesListEntry
    data-test="serid-2"
    key="serid-2"
    metadata={
      Object {
        "$schema": "https://127.0.0.1:5000/schemas/series/series-v1.0.0.json",
        "abstract": "This is a multipart monograph",
        "authors": Array [
          "Banek, Cora",
          "Banek, Georg",
        ],
        "mode_of_issuance": "MULTIPART_MONOGRAPH",
        "pid": "serid-2",
        "title": "The Gulf: The Making of An American Sea",
      }
    }
    volume={null}
  />
  <SeriesListEntry
    data-test="serid-3"
    key="serid-3"
    metadata={
      Object {
        "$schema": "https://127.0.0.1:5000/schemas/series/series-v1.0.0.json",
        "abstract": "This is serial",
        "mode_of_issuance": "SERIAL",
        "pid": "serid-3",
        "title": "Springer tracts in modern physics",
      }
    }
    volume={null}
  />
</ItemGroup>
`;
